### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by Nikolassparrow**
Something that Legends and I talked about a few weeks back was the need for the project becoming more accessible to new eyes, and the first step in doing that is providing documentation. 

This PR is a bunch of documentation on most files in the H1Z1 server to the best of my knowledge. I didn't bother to touch functions as most are self-documented or a clutter of sloppy code, so I just bothered with the ambiguous variables instead :D

Also while implementing this, I noticed that it was very common to have multiple names all meaning the same thing: the most common was probably model id's with: "actorModelId, modelIds, actorId" etc. Also was common to see some unused variables, perhaps they'll be used in the future..?

If I didn't get some information correct on a certain aspect of the project PLEASE change it! I don't want incorrect information to be presented.

### CODE REVIEW

**Comment by QuentinGruber** [comment]
i'm reviewing it it's just huge so it takes time

File: src/servers/ZoneServer2016/entities/constructiondoor.ts [hunk 1]
```diff
@@ -55,18 +55,39 @@ function getMaxHealth(itemDefinitionId: Items): number {
...
+  /** ItemDefinitionId of the door */
   readonly itemDefinitionId: number;
+
+  /** Current slot from the parent object the door is attached to */
   readonly slot: string;
+
+  /** Range at which the door entity will receive damage from explosions */
   damageRange: number;
+
+  /** Fixed position the door remains at whilst closed */
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/entities/constructiondoor.ts:79]:
```suggestion
  /** Fixed position the door remains at while it’s closed */
```

File: src/servers/ZoneServer2016/entities/vehicle.ts [hunk 2]
```diff
@@ -163,33 +164,61 @@ function getHotwireEffect(vehicleId: VehicleIds) {
 }
 
 export class Vehicle2016 extends BaseLootableEntity {
+  /** Used for determining seat swap logic*/
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/entities/vehicle.ts:167]: Not really if i’m not mistaken it’s to know if the vehicle physics are currently managed by a client

File: src/servers/ZoneServer2016/entities/vehicle.ts [hunk 3]
```diff
@@ -163,33 +164,61 @@ function getHotwireEffect(vehicleId: VehicleIds) {
...
   positionUpdate: any /*positionUpdate*/;
+
+  /** Speed (H1Z1 rpms) of the vehicle */
   engineRPM: number = 0;
+
+  /** Used by resources to determine the fuel level */
   fuelUpdater: any;
+
+  /** Returns true if the player is spectating or is parachuting
+   * (spectating is treated as being inside a vehicle) */
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/entities/vehicle.ts:199]: An hacky aircraft to use the terms of h1z1 devs 😄

File: src/servers/ZoneServer2016/zoneserver.ts [hunk 4]
```diff
@@ -254,6 +257,7 @@ const spawnLocations2 = require("../../../data/2016/zoneData/Z1_gridSpawns.json"
   > = require("../../../data/2016/sampleData/equipmentModelTexturesMapping.json");
 
 export class ZoneServer2016 extends EventEmitter {
+  /** Networking/Protocol related */
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/zoneserver.ts:260]: Networking layer allowing us to send game data to the gameclient

Comment (QuentinGruber) [reply src/servers/ZoneServer2016/zoneserver.ts:260]: This layer is on top of the h1z1protocol so h1z1 game packets

**Review by QuentinGruber — CHANGES_REQUESTED** [review]

**Comment by Nikolassparrow** [comment]
Will look into this today 👍

File: src/servers/ZoneServer2016/classes/smeltingentity.ts [hunk 5]
```diff
@@ -61,36 +61,52 @@ function getSmeltingEntityData(
...
+  workingEffect: number = Effects.EFX_Fire_Furnace;
+
+  /** Returns true when the smelting entity has been ignited */
   isWorking: boolean = false;
+
+  /** Returns true when the item is smelted */
   isSmelting: boolean = false;
-  smeltingTime: number = 40000;
+
+  /** Global dictionary of SmeltingEntities - uses CharacterId (string) for indexing */
```
[the remark this answers is not in the document]
Comment (Nikolassparrow) [reply src/servers/ZoneServer2016/classes/smeltingentity.ts:104]: Either or, Could also include a (hashmap) next to dictionary for some clarity?

File: src/servers/ZoneServer2016/entities/constructiondoor.ts (hunk 1 above)
Comment (Nikolassparrow) [reply src/servers/ZoneServer2016/entities/constructiondoor.ts:79]: hahaha that's just a 🤓🤓 english word, whilst = while it's

Comment (QuentinGruber) [reply src/servers/ZoneServer2016/entities/constructiondoor.ts]: ahah never saw it

**Comment by QuentinGruber** [comment]
there  are still conflicts will review this after conflicts are fixed

File: src/servers/ZoneServer2016/entities/baseentity.ts [hunk 6]
```diff
@@ -14,54 +14,75 @@
...
   actorModelId!: number;
+
+  /** State of the BaseLightweightCharacter, includes: state (Float32Array),
+   * rotation(Float32Array), lookAt(Float32Array), and yaw (number) */
   state: {
     position: Float32Array;
     rotation: Float32Array;
   };
+
+  /** Physical size of the entity */
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/entities/baseentity.ts:71]:
```suggestion
  /** Physical size of the entity based on the entity model */
```

File: src/servers/ZoneServer2016/entities/basefullcharacter.ts [hunk 7]
```diff
@@ -49,19 +51,22 @@ const loadoutSlots = require("./../../../../data/2016/dataSources/LoadoutSlots.j
 
 function getGender(actorModelId: number): number {
   switch (actorModelId) {
-    case 9510: // zombiemale
-    case 9240: // male character
+    case ModelIds.ZOMBIE_FEMALE_WALKER: // zombiemale
```
[the remark this answers is not in the document]
Comment (QuentinGruber) [reply src/servers/ZoneServer2016/entities/basefullcharacter.ts]: well it wasn't you

File: data/2016/zoneData/Z1_POIs.json [hunk 8]
```diff
@@ -58,18 +58,10 @@
     "range": 200,
     "bounds": [
       [
-        [ -802.030029296875, -1885.239990234375 ],
-        [ -577.3099975585938, -1889.8499755859375 ],
-        [ -577.2699890136719, -1423.489990234375 ],
-        [ -802.4400024414062, -1423.9599609375 ]
-      ]
-    ],
-    "shackBounds": [
```
[the remark this answers is not in the document]
Comment (QuentinGruber) [reply data/2016/zoneData/Z1_POIs.json:67]: beside this it's good for me

[the remark this answers is not in the document]
Comment (Nikolassparrow) [reply data/2016/zoneData/Z1_POIs.json:67]: Oh wtf, I don't know why it did that, I don't want it removed of course haha

**Review by QuentinGruber — APPROVED** [review]

**Pull request merged**